cpp/src/io/parquet/parquet_reader.cu

```diff
@@ -600,26 +549,25 @@ void decode_page_data(
   }
 }
 
-/**
- * @brief Reads Apache Parquet data and returns an array of gdf_columns.
- *
- * @param[in,out] args Structure containing input and output args
- *
- * @return gdf_error GDF_SUCCESS if successful, otherwise an error code.
- **/
-gdf_error read_parquet_arrow(pq_read_arg *args, std::shared_ptr<arrow::io::RandomAccessFile> file_handle) {
-
-  int index_col = -1;
-  DataSource input;
-  if(args->source_type == FILE_PATH){
-    input = DataSource(args->source);
-  }else if(args->source_type == ARROW_RANDOM_ACCESS_FILE){
-    CUDF_EXPECTS(file_handle != nullptr,"read_parquet_arrow was used but no file handle set");
-    input = DataSource(file_handle);
-  }else{
-    input = DataSource(args->source, args->buffer_size);
-  }
+//
+// Implementation for reading Apache ORC data and outputting to cuDF columns.
+//
+gdf_error read_parquet_arrow(
+    pq_read_arg *args,
+    std::shared_ptr<arrow::io::RandomAccessFile> rd_file) {
+
+  auto createDataSource = [&]() {
+    if (args->source_type == FILE_PATH) {
+      return DataSource(args->source);
+    } else if (args->source_type == ARROW_RANDOM_ACCESS_FILE) {
+      CUDF_EXPECTS(rd_file, "Invalid Arrow file handle");
+      return DataSource(rd_file);
+    } else {
+      return DataSource(args->source, args->buffer_size);
+    }
+  };
 
+  DataSource input = createDataSource();
```

Comment:
Not sure if you would consider this more or less readable, but you can declare and call the initialization lambda inline with DataSource declaration. Something like this: https://www.bfilipek.com/2016/11/iife-for-complex-initialization.html

Reply:
Cool; changed it to save a couple of lines thanks!